ingests_tracker/src/main/scala/uk/ac/wellcome/platform/storage/ingests_tracker/services/CallbackNotificationService.scala

```diff
@@ -0,0 +1,40 @@
+package uk.ac.wellcome.platform.storage.ingests_tracker.services
+
+import java.net.URI
+
+import grizzled.slf4j.Logging
+import uk.ac.wellcome.json.JsonUtil._
+import uk.ac.wellcome.messaging.MessageSender
+import uk.ac.wellcome.platform.archive.common.ingests.models.Callback.Pending
+import uk.ac.wellcome.platform.archive.common.ingests.models.{
+  Callback,
+  CallbackNotification,
+  Ingest
+}
+
+import scala.util.{Success, Try}
+
+class CallbackNotificationService[Destination](
+  messageSender: MessageSender[Destination]
+) extends Logging {
+  def sendNotification(ingest: Ingest): Try[Unit] =
+    ingest.callback match {
```

Comment:
Could this be done neatly without the nested matches?

Reply:
This code is copied verbatim from the ingest service (along with its' tests) rather than rewritten so as to reduce the possibility of me introducing errors. 

I will fix this though as we've quit good test coverage.